arkit/server.py

```diff
@@ -0,0 +1,256 @@
+                if self._round == 3:
+                    self._round = 0
+
+    async def find_cube(self):
+        cubes = None
+        if self._cube:
+            self._cube.stop_light_chaser()
+            self._cube.set_lights_off()
+        look_around = self._robot.start_behavior(cozmo.behavior.BehaviorTypes.LookAroundInPlace)
+        try:
+            cubes = await self._robot.world.wait_until_observe_num_objects(1, cozmo.objects.LightCube)
+            self._cube = cubes[0]
+            look_around.stop()
+        except TimeoutError:
+            print("Didn't find a cube :-(")
+            return False
+        if self._round < 2:
+            tap_anim = random.choice(["anim_pyramid_reacttocube_happy_high_01", "anim_speedtap_winround_intensity02_02"])
+            await self._robot.play_anim(tap_anim).wait_for_completed()
+        if self._round > 2:
+            return
+        self._cube.start_light_chaser(GREEN)
+        await self._robot.set_head_angle(cozmo.util.Angle(degrees=20.5)).wait_for_completed()
+        await self._robot.set_head_angle(cozmo.util.Angle(degrees=0)).wait_for_completed()
+        if self._cube:
+            await self.go_to_cube(self._cube)
+
+    async def go_to_cube(self,cube):
+        await self._robot.set_lift_height(1, duration=0.5).wait_for_completed()
+        await self._robot.go_to_object(cube, distance_mm(30.0)).wait_for_completed()
+        self._wait_for_tap_flag = True
+        await self.coz_tap(cube)
+
+    async def coz_tap(self,cube):
+        if not self._wait_for_tap_flag:
+            return
+        threading.Timer(0.5, self.trigger_cozmo_tap).start()
+        tap_anim = random.choice(["anim_speedtap_tap_01", "anim_speedtap_tap_02", "anim_speedtap_tap_03"])
+        await self._robot.play_anim(tap_anim).wait_for_completed()
+        await self.see_fireworks(cube)
+
+    async def see_fireworks(self,cube):
+        if self._round != 2:
+            await self._robot.drive_straight(distance_mm(-50), speed_mmps(80)).wait_for_completed()
+            await self._robot.set_lift_height(0, duration=0.3).wait_for_completed()
+
+        messages = [1,2,3]
+        message = messages[self._round]
+
+        # Tap unsuccessful
+        if self._animation_triggerd is False:
+            self._trigger_message = False
+            await self._robot.play_anim("anim_keepaway_losehand_01").wait_for_completed()
+
+        # Cozmo taps cube and looks at fireworks
+        if self._animation_triggerd and self._round < 2:
+            self.send_msg(message)
+
+            # Cozmo acts excited
+            if self._round == 0:
+                await self._robot.set_head_angle(cozmo.util.Angle(degrees=44.5),1,1,1).wait_for_completed()
+                await asyncio.sleep(1)
+                await self._robot.play_anim("anim_reacttoblock_react_01_head_angle_40").wait_for_completed()
+                await self._robot.drive_straight(distance_mm(-30), speed_mmps(30)).wait_for_completed()
```

Comment:
As per above, use of play_anim with specific animation name is fragile and could break with a future app version.